### PR TITLE
join methods for tbl_spark

### PR DESCRIPTION
**Pull request by wkdavis**
@yl790 It looks like the default test version of Spark is 2.3.0, but I can only run >= 3.0.0 on my machine due to my version of Java, so I will watch the builds to make sure it passes.

### CODE REVIEW

File: R/dplyr_join.R [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+#' with underscores (`_`). See [join] for a description of the general purpose
+#' of the functions.
+#'
+#' @inheritParams dbplyr::join.tbl_sql
+#'
+#' @name join.tbl_spark
+NULL
+
+#' @rdname join.tbl_spark
+#' @export
```
Comment (yl790) [inline R/dplyr_join.R:14]: Minor: currently the entry for this method is `export(inner_join.tbl_spark)` in the NAMESPACE file, which is a bit off from the truth.

I think if you make the suggested change below and re-run `roxygen2::roxygenize()`, the NAMESPACE entry will become `S3method(inner_join,tbl_spark)`, which is more technically correct.

```suggestion
#' @importFrom dplyr inner_join
#' @export
```

Comment (wkdavis) [reply R/dplyr_join.R:14]: Agreed, should be an s3 method. I made the changes - it required reexporting the generics from `dplyr` in order to register the s3 method (I tried using just the `importFrom` block as you suggested but that did not register the methods).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2651-1/8bc192e3-743b-4f2d-8778-5a4cd4fc6fff.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2651-2/aa24d984-4c69-4940-aee8-9a511a5bd84a.txt).

**Pull request merged**